File: materials/2019-2020/11v/2019-12-18-polymorphism-exercise/point.cc
#include "point.hh"

Point::Point(double x, double y): x(x), y(y) {}

double Point::get_x() const {
  return x;
}

double Point::get_y() const {
  return y;
}

std::ostream& operator<<(std::ostream& os, Point const& point) {
  return os
    << '('
    << point.get_x()
    << ", "
    << point.get_y()
    << ')';
}

File: materials/2022-2023/11a/2022-11-04-shapes/Shapes/src/Point.java
public class Point {
    private int x;
    private int y;

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int getX() {
        return x;
    }


    public int getY() {
        return y;
    }


    public static double getSide(Point a, Point b)
    {
        return Math.sqrt(Math.pow((a.getX() - b.getX()), 2) + Math.pow((a.getY() + b.getY()), 2));
    }

    public static PointHoader max(Point a, Point b, Point c)
    {
        Point temp = a;
        PointHoader otherPoints = null;

        if(temp.getY() <= b.getY())
        {
            temp = b;
            otherPoints = new PointHoader(temp, a, c);
        }


        if(temp.getY() <= c.getY())
        {
            temp = c;
            otherPoints = new PointHoader(temp, b, a);
        }

        if(null == otherPoints)
            otherPoints = new PointHoader(temp, b, c);

        return otherPoints;
    }
}
